fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/BlobStorePayload.java

```diff
@@ -45,20 +62,22 @@ public BlobStorePayload(int resourceTypeId, String logicalId, int version, Strin
      * @param client
      * @throws FHIRPersistenceException
      */
-    public Response<BlockBlobItem> run(BlobManagedContainer client) throws FHIRPersistenceException {
-        final StringBuilder blobNameBuilder = new StringBuilder();
-        blobNameBuilder.append(resourceTypeId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(logicalId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(version);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(resourcePayloadKey);
-        BlobClient bc = client.getClient().getBlobClient(blobNameBuilder.toString());
+    public CompletableFuture<PayloadPersistenceResult> run(BlobManagedContainer client) throws FHIRPersistenceException {
         
-        BlobParallelUploadOptions uploadOptions = new BlobParallelUploadOptions(BinaryData.fromBytes(ioStream.getRawBuffer()));
-        Response<BlockBlobItem> response = bc.uploadWithResponse(uploadOptions, client.getProperties().getTimeout(), null);
-        
-        return response;
+        if (ioStream.size() > MAX_APPEND_BLOB_BLOCK_SIZE) {
+            throw new FHIRPersistenceException("Resource payload size cannot exceed " + MAX_APPEND_BLOB_BLOCK_SIZE + " bytes");
+        }
+        final String blobPath = BlobPayloadSupport.getPayloadPath(resourceTypeId, logicalId, version, resourcePayloadKey);
+        logger.fine(() -> "Payload storage path: " + blobPath);
+        BlobAsyncClient bc = client.getClient().getBlobAsyncClient(blobPath);
+
+        // Reactor pipeline to first call create then perform the upload.
+        return bc.getAppendBlobAsyncClient()
+                .create(true)
+                .flatMap(item -> {
+                    return bc.getAppendBlobAsyncClient().appendBlockWithResponse(Flux.just(ioStream.wrap()), ioStream.size(), null, null);
```

Comment:
this bit looks a bit foreign to me...

Reply:
Yeah, the async API stuff takes a bit of getting used to. I'll add a comment to explain what's going on here.